src/js/components/AppVersionComponent.jsx

```diff
@@ -187,8 +188,15 @@ var AppVersionComponent = React.createClass({
 
     var urisNode = (appVersion.uris == null || appVersion.uris.length === 0)
       ? <UnspecifiedNodeComponent />
-      : appVersion.uris.map(function (u) {
-        return <dd key={u}>{u}</dd>;
+      : appVersion.uris.map(function (uri, i) {
+        var parsedURI = url.parse(uri);
+        var linkName = uri;
```

Comment:
I really like your suggestion, like this we can get one step closer to the KingdomOfNouns